app/src/main/java/Battleship/HelpFunctions.java

```diff
@@ -144,7 +144,7 @@ public static boolean generalCheckOfTheShip(String [] fourDeck, int n, String []
 		
 		boolean allIsOk = true;
 		
-		allIsOk = allIsOk && HelpFunctions.shipLengthCheck(fourDeck, n);  // если после какого-то метода allIsOk станет false, что дальше строки не будут читаться, т.к. из false уже никогда не станет true
+		allIsOk = allIsOk && HelpFunctions.shipLengthCheck(fourDeck, n);  // if after the execution of some method allIsOk becomes false, then further lines will not be read, because false will never become true
```

Comment:
Wrong. Second operand (method) of && will not be call. Because first operand are already false.
All lines will be read